### DISCUSSION TITLE
Help with Pico/RP2040 and ILI4681/ILI4688

### DISCUSSION THREAD

**Discussion by tomboblombo** [Bodmer/TFT_eSPI#2021] [Q&A - General] [answered]
Hey everyone,

I'm trying to get my Pico/RP2040 to display to either a ILI4681 or ILI4688. I've tried everything I can think of, but I haven't gotten a single thing to show up yet.

These are the screens I'm using:
https://www.aliexpress.com/item/3256803594335940.html (ILI9481 No Touch)
https://www.aliexpress.com/item/3256801812981724.html (ILI9488 No Touch)

This is my user setup file: 
[User_Setup.txt](https://github.com/Bodmer/TFT_eSPI/files/9568922/User_Setup.txt)

And my pin connections are as follows:
Pico / TFT
GND -> GND
3V3(OUT) -> VDD, BL
GP2 -> SCK
GP3 -> SDI
GP18 -> D/C
GP19 -> RST
GP20 -> CS

![PXL_20220914_181900163](https://user-images.githubusercontent.com/5630030/190234902-7bf5fb16-25cc-4df7-a4bb-bcd518fbd26d.jpg)

I've tried a variety of the built-in examples, and I'm compiling/uploading via Arduino using https://github.com/earlephilhower/arduino-pico/ for Pico support.

Any help would be greatly appreciated... Thanks!

**Comment by Bodmer** [comment]
Run the "Read_User_Setup" test and diagnostic example. Nothing is displayed on the screen but it will report the setup being used by the compiler. Check this against the expected values.

The setting of 70MHz for SPI clock is too high for those screens. Start at 10MHz and see if that works.

**Reply by tomboblombo** [reply]
So, I changed the clock frequency down to 10MHz and ran the Read_User_Setup sketch. The values look like what I want: [SerialOutput.txt](https://github.com/Bodmer/TFT_eSPI/files/9580727/SerialOutput.txt)

Unfortunately, I'm still not getting any visuals on either screen after uploading the 'Ellipse' example onto the board...

Anything else I can test?

**Reply by tomboblombo** [reply]
Also, if there's a known hardware setup that works, I'd be willing to test that. I think a possibility is that screens are duds...

**Reply by tomboblombo** [reply]
I'm still trying different things to no avail. I've tried a few different board types to compile to code to the pico, such as Arduino's supplied RP2040 mbed board.

**Reply by KBLiveSolutions** [reply]
Did you make any progress ?
I'm having similar behavior with a RP2040 and a ILI9486 board [(Waveshare 4inch TFT)](https://www.waveshare.com/4inch-tft-touch-shield.htm)
Read_User_Setup looks fine, TFT_ReadWrite_Test gives me only errors.

**Reply by tomboblombo** [reply]
Unfortunately, no. I've moved onto using this board (similar to the working setup listed below), but I've still had no luck even using their example documentation...
https://www.waveshare.com/wiki/Pico-ResTouch-LCD-3.5

I've also tested all my displays using an Arduino Nano Every, Seeeduino Xiao, and Teensy 4.0 and none of them have worked with either my ILI4681/ILI4688.

I'm not sure if the library doesn't work at all, or if every single one of my displays or dev boards are broken... 😖

**Comment by Bodmer** [comment, marked as answer]
Working setup here that may help.
https://github.com/DustinWatts/Pico-Matrix-Touch-Keyboard

**Reply by tomboblombo** [reply]
Thanks!

I ended up springing for this, as it uses the same parts, and accomplishes the same goal.
https://www.waveshare.com/wiki/Pico-ResTouch-LCD-3.5

Unfortunately, even following their documentation (or the touch kb documentation), I've had no luck getting the screen working...

**Comment by TangerineDreamer** [comment]
I've done the setup bodmer that suggested and i did it working the first time :)
but had a bunch of errors with the Littlefs : FS.h not found ,as i did understand,it's called now "lfs" with the new lib 2.05
and i tried to replace with "#define FILESYSTEM lfs":
//-------------------------------------------------
#include <lfs.h>

#include <SPI.h>
#include <TFT_eSPI.h>

// Name of the Touch calibration file
#define CALIBRATION_FILE "/TouchCalData2"

// Define our filesystem
#define FILESYSTEM lfs
//-----------------------------------and so on-----------------------------------


but had strange errors like that:


 error: expected primary-expression before '.' token
   if (FILESYSTEM.exists(CALIBRATION_FILE)){

It is the exact copy than the SPIFFS code,don't understand the source of this error.

**Comment by TangerineDreamer** [comment]
It's a shame neither of them work,
i looked at your setup,you don't use the same for D/C and RST,you could try the exact setup proposed for these pins:
D/C 22 & RST 21,because the PICO has maybe different data treatments on those pins,i use RP2040 black,not the green one
but i think they are the same,same pins...
why not try with RST -1 (wired to the 3,3v)
have you got the last pico board package from earlephilhower - (https://github.com/earlephilhower/arduino-pico) ?

**Reply by tomboblombo** [reply]
Thanks for the ideas. I'll try them after work today.

**Comment by tomboblombo** [comment]
Hey-o,

Good news. I now have a working setup with the Waveshare 3.5" module https://www.waveshare.com/wiki/Pico-ResTouch-LCD-3.5
That means I can finally start branching out with a stable base. 👌

The reason I was having trouble setting my boards up recently was because at some point I had copied my "User_Setup_select.h" file to the "User_Setups" directory and was modifying it instead. Only discovered after running the "Read_User_Setup" sketch again.

Side note, but is there a way to define a user setup in the local application? I don't really like having them in the libraries folder...

I'll try with other boards again and report back with any other successful setups.

**Reply by KBLiveSolutions** [reply]
Con you please share your working setup?
I should receive this display tomorrow, so the timing is perfect.

You can add a setup file in your sketch, it's detailed in the [Sketch_with_tft_setup](https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/Sketch_with_tft_setup) example

**Reply by tomboblombo** [reply]
The setup is described on Waveshare's wiki. Don't think there's much for me to add.
